src/api_v2/api.rs

```diff
@@ -0,0 +1,36 @@
+use std::collections::HashMap;
+use std::fmt::format;
+use std::sync::Arc;
+use bytes::Bytes;
+use reqwest::header::HeaderMap;
+use crate::api_v2::{BeatMap, Error, User};
+
+pub struct Api {
+    bot: Arc<User>,
+    client: reqwest::Client,
+}
+
+impl Api {
+    pub fn new(client_id: String, code: String, url: String) -> Result<Self, Error> {
+        let mut bot = User::new_bot(code);
```

Comment:
Can we use `User::new_bot(code, client_id)` here?

Reply:
yes